### ISSUE TITLE
Northstar's List and ListItems are being narrated as 'clickable' by NVDA 

### ISSUE THREAD

**Issue by davyrwoo** [microsoft/fluentui#20658]
### Environment Information

- **Package version(s)**: 0.59.0

### Describe the issue:

When Northstar `<List>` and `<ListItem>` do not have the `selectable` or `navigable` props, they are not clickable. When using NVDA in browse mode, navigating to the `<List>` and `<ListItems>` narrates as "clickable"

### Please provide a reproduction of the issue in a codepen:

https://codesandbox.io/s/3hrwi?module=/example.js

Navigate to the List with NVDA in browse mode

#### Actual behavior:

NVDA narrates the items as "clickable"

#### Expected behavior:

These items are not clickable, so NVDA should not announce that it is

**Comment by gouttierre** [comment]
@davyrwoo - Thanks for filing this issue with us. To set expectations, the developers will review this issue once capacity allows.

@jurokapsiar- Would your team be able to confirm if this is a regression, or if this behavior is an issue?
